stdlib/Dates/src/deprecated.jl

```diff
@@ -68,3 +68,17 @@ end
 
 @deprecate argerror(msg::String) ArgumentError(msg) false
 @deprecate argerror() nothing false
+
+@noinline function (+)(x::DateTime, y::Union{Microsecond, Nanosecond})
+    # throw(InexactError(:+, Millisecond, y))
+    # TODO: eventually just throw, and move to arithmetic.jl
+    Base.depwarn("adding DateTime and Microsecond (or Nanosecond) is deprecated.", :+)
+    return x + Millisecond(toms(y))
+end
```

Comment:
According to the discussion above

```suggestion
@noinline function (+)(x::DateTime, y::Nanosecond)
    # throw(InexactError(:+, Millisecond, y))
    # TODO: eventually just throw, and move to arithmetic.jl
    v = value(y)
    !iszero(v % tons(Millisecond(1)) && Base.depwarn("adding DateTime and Microsecond (or Nanosecond) is deprecated.", :+)
    return x + Millisecond(toms(y))
end
```

and similarly then for `Microsecond`, as well as `-`. Unfortunately, there is no `Dates.toµs` equivalent.

Reply:
Yes, that ought to work too.